var/spack/repos/builtin/packages/paraview/package.py

```diff
@@ -412,6 +413,13 @@ def nvariant_bool(feature):
         cmake_args.append(
             '-DPARAVIEW_BUILD_SHARED_LIBS:BOOL=%s' % variant_bool('+shared'))
 
+        # VTKm added to paraview in 5.3.0 and up
+        if spec.satisfies('@5.3.0:'):
+            if '~vtkm' in spec:
+                cmake_args.append('-DPARAVIEW_USE_VTKM:BOOL=OFF')
+            else:
+                cmake_args.append('-DPARAVIEW_USE_VTKM:BOOL=ON')
```

Comment:
Is it possible something as

```suggestion
        # VTKm added to paraview in 5.3.0 and up
        if spec.satisfies('@5.3.0:') and spec['vtkm'].value != speck['value'].default:
            if '+vtkm' in spec:
                cmake_args.append('-DPARAVIEW_USE_VTKM:BOOL=ON')
            else:
                cmake_args.append('-DPARAVIEW_USE_VTKM:BOOL=OFF')
```

Reply:
@vicentebolea Wouldn't this make building `canonical` without `vtkm` impossible? `spack install paraview ~vtkm` would do nothing since `~vtkm` is the default? Also, the default paraview spec would show `~vtkm` even though it actually is building vtkm.

Reply:
I am not very sure about this but from their reference readthedocs it suggest that default=False is different to value (which is + or ~). I am not strongly versed in Spack so I might be wrong. But the idea is that if we do not specify the vtkm variant we just do not either enable of disable `PARAVIEW_USE_VTKM`

Reply:
Oh cool. That could work. But I do feel like there could be weirdness with `spack install paraview` and `spack install paraview ~vtkm` having the same spec with different results. I'll have to try it out.

Reply:
The idea is to make spack use the Cmake package as an user as much as possible and not trying to replicate the internal logic of how the CMake package internally setup its options. The latter would then add an indirect dependency

Reply:
What if it's a multi option, like the build_edition? Default would be something like `use_vtkm=default`, which doesn't change anything. Then it can be enabled with something like `use_vtkm=on` and disabled with `use_vtkm=off`.

Reply:
Good idea, like this we can explicitly declare and control the three states {DEFAULT, ON, OFF}